src/typescript/09-interfaces/interfaces.md

```diff
@@ -1,78 +1,197 @@
+}
+```
 @codepen
 
-## Interfaces in Functions
+### Interfaces in functions
 
 Interfaces are incredibly useful in describing the shape of objects we want to use in multiple situations. The following functions both require a ``Dinosaur`` object shape we’ve defined in the ``Dinosaur`` interface.
 
-@sourceref ./7-3-2-functions.ts
+```typescript
+interface Dinosaur {
+  name: string;
+  breed: string;
+  location: string;
+}
+
+let dinoA = {
+  name: 'Blue',
+  breed: 'Velociraptor',
+  location: 'Section B'
+};
+
+let dinoB = {
+  name: 'Sally',
+  location: 'Section C'
+};
+
+function dinoCatcher(dinosaur: Dinosaur) {
+  console.log(`Caught ${dinosaur.name} at ${dinosaur.location}`);
+}
+
+dinoCatcher(dinoA);
+//works!
+dinoCatcher(dinoB);
+//Argument of type '{ name: string; location: string; }' is not assignable to parameter of type 'Dinosaur'.
+//Property 'breed' is missing in type '{ name: string; location: string; }'.
+```
 @codepen
 
-## Interfaces Describing Functions
+### Interfaces describing functions
 
 We can also use interfaces to describe functions, basically creating reusable types for functions. On the left side in parenthesis we list the parameters, and to the right of the colon we state the return type.
 
-@sourceref ./7-3-3-describefuncs.ts
+```typescript
+interface DinoDNAMixer {
+  (dino1: string, dino2: string, spliceIdx: number): string;
+}
+
+let dinoMaker : DinoDNAMixer = function (dino1: string, dino2: string, spliceIdx: number): string {
+  return dino1.substring(spliceIdx) + dino2.substring(spliceIdx);
+}
+
+let newDino = dinoMaker('CGGCAD', 'ACGCAA', 3);
+console.log(newDino); //logs 'CADCAA'
+```
 @codepen
 
 It’s possible to use the `type` keyword as an interface to describe a function.
 
-@sourceref ./7-3-4-describefuncs-type.ts
+```typescript
+type DinoDNAMixer = (dino1: string, dino2: string, spliceIdx: number) => string;
+
+interface DinoFactory {
+  makeDino: DinoDNAMixer;
+  factoryName: string;
+}
+```
 @codepen
 
-## Type Assertion
+### Type assertion
 
 We briefly mentioned type assertion when talk about types, but when dealing with interfaces it can be a great tool for making sure our code behaves in the way we expect.
 
 For instance, consider the following code:
 
-@sourceref ./7-4-type-assertion.ts
+```typescript
+interface Dinosaur {
+  name: string;
+  height: number;
+}
+
+var myObj = {} as Dinosaur;
+myObj.name = 'Blue';
+myObj.height = 6;
+```
 @codepen
 
 When we create empty object literals in TypeScript, they are inferred to be objects with zero properties. To fix this, we can use type assertions to let the compiler explicitly know what we want from our object.
 
-@sourceref ./7-4-2-type-assertion.ts
+```typescript
+var myObj = {};
+myObj.name = 'Blue';
+//property 'name' does not exist on type '{}'
+myObj.height = 6;
+//property 'height' does not exist on type '{}'
+```
 @codepen
 
-## Exercise 1: Write Interfaces to Describe an Object
+### Exercise 1: Write Interfaces to Describe an Object
```

Comment:
It looks like this first exercise only covers the first two concepts (through optional properties), so this exercise should be after those two concepts and then a new objective should start after that.